ownCloud/Client/ClientQueryViewController.swift

```diff
@@ -226,6 +229,8 @@ class ClientQueryViewController: UITableViewController, Themeable {
 		// Make sure we don't request the thumbnail multiple times in that case.
 		if cell?.item?.versionIdentifier != newItem.versionIdentifier {
 			cell?.item = newItem
+		} else {
+			cell?.updateWith(newItem)
```

Comment:
I assume this is about the issue you noticed where file name changes don't change eTag and fileID and therefore don't update the cell. With this change, however, the cell would always be updated (cell?.item = .. calls through to updateWith(item) as well).

To update the cell only if the contents or the filename changed, something like this should be sufficient:
```swift
if (cell?.item?.versionIdentifier != newItem.versionIdentifier) || (cell?.item?.name != newItem.name) {
   cell?.item = newItem
}
```